ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/KsqlKeyExtractor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import io.confluent.ksql.GenericRow;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class KsqlKeyExtractor<KRightT> implements Function<GenericRow, KRightT> {
```

Comment:
Just a question here: I saw different Extractor classes in the repo, and some with a `Ksql` prefix in its name and some do not. Is there a convention for the naming at the moment? cc @agavra @vcrfxia

Reply:
🔥 ⚠️ - I really don't like `Ksql` prefix. I don't think we have a convention, so it's really just do what you like.